### PR TITLE
Feat 32 code ocean registration

### PR DESCRIPTION
**Pull request by jtyoung84**

### CODE REVIEW

**Comment by jtyoung84** [comment]
Closes #32 
Closes #38 

Allows user to optionally define the raw_data_source folder using a command line argument since that option is changed more often than the other configs.

**Comment by jtyoung84** [comment]
I added some basic unit tests for the post request to code ocean, but we may want to figure out a more robust method for testing out that process.

File: src/transfer/jobs/openephys_job.py [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
-        args.conf_file_location
-    )
+    job_configs = EphysJobConfigurationLoader().load_configs(sys.argv[1:])
 
     # Extract raw data name, (e.g., openephys) and raw data path
     data_name = job_configs["data"]["name"]
-    data_src_dir = Path(job_configs["data"]["source_dir"])
-    data_dest_dir = Path(job_configs["data"]["dest_dir"])
+    data_src_dir = Path(job_configs["endpoints"]["raw_data_dir"])
+    shrunk_data_dir = Path(job_configs["endpoints"]["shrunk_data_dir"])
```
Comment (alejoe91) [inline src/transfer/jobs/openephys_job.py:21]: Should this be `clipped` instead of `shrunk`?

File: src/transfer/configuration_loader.py [hunk 2]
```diff
@@ -1,11 +1,25 @@
...
+
 import yaml
 from numcodecs import Blosc
 
 
 class EphysJobConfigurationLoader:
     """Class to handle loading ephys job configs"""
 
+    class RegexPatterns(Enum):
+        """Enum for compression algorithms a user can select"""
```
Comment (alejoe91) [inline src/transfer/configuration_loader.py:16]: docstrings look off here

**Review by alejoe91 — CHANGES_REQUESTED** [review]
@jtyoung84 a couple of suggested changes

File: conf/ephys_upload_job_configs.yml [hunk 3]
```diff
@@ -20,8 +30,9 @@ compress_data_job:
...
 upload_data_job:
-  dryrun: false # Set this as false to perform the actual upload
-  upload_to_s3: true # Set as false to skip uploading to s3
-  s3_dest:  # Set to something like s3://BUCKET/PATH
-  upload_to_gcp: true # Set as false to skip uploading to s3
-  gcp_dest:  # Set to something like gs://BUCKET/PATH
+  dryrun: false # Set this to true to perform a dryrun of the upload
+register_on_codeocean_job:
+  api_token: # Defaults to ENV Variable CODEOCEAN_API_TOKEN
+  tags: ['ecephys']
```
Comment (dyf) [inline conf/ephys_upload_job_configs.yml:36]: can we add a `raw` tag here?

File: src/transfer/configuration_loader.py (hunk 2 above)
Comment (jtyoung84) [reply src/transfer/configuration_loader.py]: Nice catch. I updated it.

File: tests/resources/test_configs/ephys_upload_job_test_configs.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    kwargs: {"shuffle": "BITSHUFFLE"}
+  scale_params:
+    num_chunks_per_segment:
+    chunk_size: 20
+    disable_tqdm:
+upload_data_job:
+  dryrun: true # Set this to true to perform a dryrun of the upload
+register_on_codeocean_job:
+  api_token: # Defaults to ENV Variable CODEOCEAN_API_TOKEN
+  tags: ['ecephys']
```
Comment (dyf) [inline tests/resources/test_configs/ephys_upload_job_test_configs.yml:36]: add `raw` tag here too

**Review by dyf — COMMENTED** [review]
Minor comments from me.  

We will probably be switching our production bucket to the open data one soon, FYI.

File: conf/ephys_upload_job_configs.yml (hunk 3 above)
Comment (jtyoung84) [reply conf/ephys_upload_job_configs.yml]: Done. In the future, we should consider extracting out aind dependent defaults from the config template to make everything more open source. Maybe we can add a config file creator script in the future?

File: tests/resources/test_configs/ephys_upload_job_test_configs.yml (hunk 4 above)
Comment (jtyoung84) [reply tests/resources/test_configs/ephys_upload_job_test_configs.yml]: done

File: src/transfer/jobs/openephys_job.py (hunk 1 above)
Comment (dyf) [reply src/transfer/jobs/openephys_job.py:21]: I think destination is fine.

File: conf/ephys_upload_job_configs.yml (hunk 3 above)
Comment (dyf) [reply conf/ephys_upload_job_configs.yml]: Yeah, that's a fine idea.

File: src/transfer/jobs/openephys_job.py (hunk 1 above)
Comment (jtyoung84) [reply src/transfer/jobs/openephys_job.py]: Renamed back to dest_data_dir

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**